File: BC - w4/java/Basic/Complex/Complex.java
package Complex;

public class Complex {
	private double real;
	private double imaginary;
	
	
	
public Complex(double real, double imaginary){
        this.real = real;
        this.imaginary = imaginary;
}
public Complex(){
    this(0,0);
}

public double getReal() {
        return this.real;
}
public double getImaginary(){
        return this.imaginary;
}

public boolean equal(Complex other) {
        return this.real == other.real && this.imaginary == other.imaginary;
}

public void increm (Complex other){
	this.real += other.real;
	this.imaginary += other.imaginary;
}

public Complex sum (Complex other){
	Complex a = this;
	a.increm(other);
        return a;
}

public void decr (Complex other){
	this.real += other.real;
	this.imaginary += other.imaginary;
}

public Complex dif ( Complex other){
	Complex a = this;
	a.decr(other);
        return a;
}


public Complex multip (Complex other) {
        Complex multip = this;

        multip.real *= other.real;
        multip.imaginary *= other.imaginary;

        return multip;
}

public void print(){
	System.out.println("("+this.real+";"+this.imaginary+"i"+")");
}

}

File: BC - w4/Basic/complex/complex.cpp
#include "complex.h"
#include <iostream>


        Complex::Complex(double real, double imaginary){
                this->real = real;
                this->imaginary = imaginary;
        };
        Complex::~Complex(){};

        double Complex::getReal() const{
                return this->real;
        };
        double Complex::getImaginary() const{
                return this->imaginary;
        };

        bool Complex::operator==(const Complex& other) const{
                return this->real == other.real && this->imaginary == other.imaginary;
        };

        bool Complex::operator!=(const Complex& other) const{
                return !operator == (other);
        };
        void Complex::operator+=(const Complex& other){
                this->real += other.real;
                this->imaginary += other.imaginary;
        };
        void Complex::operator-=(const Complex& other){
                this->real -= other.real;
                this->imaginary -= other.imaginary;
        };
        Complex Complex::operator+(const Complex& other) const{
                Complex sum = *this;

                sum += other;
                return sum;
        };
        Complex Complex::operator-(const Complex& other) const{
                Complex diff = *this;

                diff -= other;
                return diff;

        };
        Complex Complex::operator*(const Complex& other) const{
                Complex multip = *this;

                multip.real *= other.real;
                multip.imaginary *= other.imaginary;

                return multip;
        };

std::ostream& operator<<(std::ostream& out, const Complex& complex){
        out << complex.getReal();
        if (complex.getImaginary() < 0) {
        } else {
                out <<"+";
        };
        out <<complex.getImaginary()<<"i" <<std::endl;

};
